src/EditorFeatures/CSharpTest/CodeActions/IntroduceParameter/IntroduceParameterTests.cs

```diff
@@ -0,0 +1,1656 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.IntroduceVariable;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests.CodeActions;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeActions.IntroduceParameter
+{
+    public class IntroduceParameterTests : AbstractCSharpCodeActionTest
```

Comment:
darn, i wish i'd noticed this. you're using the old-school testing lib.  we have a new one (The actual public testing library we expose to customres).  This lib has the benefit that it actually validates the code you generate.  so if the code you generate produces errors, you have to mark that in the test. this helps ensure we either don't introduce errors, or we document that it's an intentional part of how that refactoring behaves.

it would have caught the issue of generating a local function overload for example.

Reply:
Can you link me a file that uses the new testing lib? I can convert the tests in a future PR?

Reply:
look for CSharpCodeRefactoringVerifier